### PR TITLE
MCOL-4180 Add some missing support for wide decimals to dbcon/execplan classes.

### PR DESCRIPTION
**Pull request by tntnatbry**

### CODE REVIEW

**Review by drrtuy — APPROVED** [review]

File: dbcon/execplan/treenode.h [hunk 1]
```diff
@@ -829,8 +836,16 @@ inline float TreeNode::getFloatVal()
             return (float)fResult.doubleVal;
 
         case CalpontSystemCatalog::DECIMAL:
+        case CalpontSystemCatalog::UDECIMAL:
```
Comment (drrtuy) [inline dbcon/execplan/treenode.h:839]: Did I get it right that previously we returned nothing for UDECIMAL?

File: dbcon/execplan/treenode.h [hunk 2]
```diff
@@ -974,8 +989,15 @@ inline long double TreeNode::getLongDoubleVal()
...
-            // this may not be accurate. if this is problematic, change to pre-calculated power array.
-            return (long double)(fResult.decimalVal.value / pow((long double)10, fResult.decimalVal.scale));
+            if (fResultType.colWidth == datatypes::MAXDECIMALWIDTH)
+            {
+                return static_cast<long double>(fResult.decimalVal);
+            }
+            else
+            {
+                // this may not be accurate. if this is problematic, change to pre-calculated power array.
+                return (long double)(fResult.decimalVal.value / pow((long double)10, fResult.decimalVal.scale));
```
Comment (drrtuy) [inline dbcon/execplan/treenode.h:999]: I would like to make this a method of the Decimal class but I think you would like to change it globally in a separate patch.

File: dbcon/execplan/treenode.h (hunk 1 above)
Comment (tntnatbry) [reply dbcon/execplan/treenode.h:839]: Yes, previously we were throwing an exception for UDECIMAL.

File: dbcon/execplan/treenode.h (hunk 2 above)
Comment (tntnatbry) [reply dbcon/execplan/treenode.h:999]: I agree, this change should be outside the scope of this patch.

**Pull request merged**